docs/prober-and-monitoring.md

```diff
@@ -0,0 +1,94 @@
+# Prober and monitoring
+
+**PMAP is not a official Google product.**
+
+Monitoring consists of two parts: prober and alerting.
+
+## Prober
+
+[Prober](../prober/) is a tool that constantly probes pmap services to check if
+the services are up. It is built as a go binary and deployed to [cloud run
+job](https://cloud.google.com/run/docs/overview/what-is-cloud-run#jobs), and is
+triggered by [cloud scheduler](https://cloud.google.com/scheduler) to constantly
+probing pmap services to check if the services are up.
+
+In each prober job, the prober will cover two CUJ:
+
+- Resource mapping (mapping service user journey)
+- Wipeout planning (policy service user journey)
+
+For each CUJ, prober will upload a yaml file to
+[gcs](https://cloud.google.com/storage), and query the bigquery table to see if
+a corrosponding entry exists in bigquery table. The job is considered as success
+on if the following requirements for both mapping and policy service are met:
+
+1. The object is uploaded.
+2. There is an entry in bigquery table that matches the traceID.
+
+## Monitoring and Alert Policies
+
+We use [pubsub
+metrics](https://cloud.google.com/monitoring/api/metrics_gcp#gcp-pubsub) to
+monitor pubsub subsrciption and sent out alerts. The metrics used are:
+
+- subscription/oldest_unacked_message_age
+- subscription/dead_letter_message_counts
+
+We also use [cloud run
+metircs](https://cloud.google.com/monitoring/api/metrics_gcp#gcp-run) to monitor
+and send alert base prober job execution result. The metric used is:
+
+- job/completed_execution_count
```

Comment:
Mention this before pubsub. And then mention why we additionally need to watch the pubsub metrics

Reply:
Added